chainlink/__init__.py

```diff
@@ -63,6 +66,32 @@ def _pull_images(self):
             if not self._pull_status.get(image):
                 raise ValueError("Failed to pull all images")
 
+    async def _pull_images_async(self):
+        images = set([stage["image"] for stage in self.stages])
```

Comment:
Most of this function seems identical to `_pull_images` with the exception of the declaration of the event loop and the running the wait call. Even the declaration of the `wait()` function can happen in both sync and async; it's just how it's called that changes.

Can we create a new function that does everything except the wait call to avoid this code duplication? It might just take the set of image names and return the wait function to call.

Reply:
I know this is all to save ~10 lines of code, but it can be helpful over time.

Reply:
Agree with the above, rest LGTM.